pyudunits2/_datetime.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+import dataclasses
+
+
+@dataclasses.dataclass(frozen=True)
+class Time:
+    hour: int
+    minute: int
+    second: int
+    microseconds: int
+
+    def __str__(self):
+        t_str = f"{self.hour:02}:{self.minute:02}"
+        if self.second or self.microseconds:
+            t_str += f":{self.second:02}"
+        if self.microseconds:
+            t_str += f".{self.microseconds:08}"
+        return t_str
+
+
+@dataclasses.dataclass(frozen=True)
+class DateTime:
+    year: int
+    month: int
+    day: int
+    time: Time
+    tz_offset: Time | str = "UTC"
+
+    # @classmethod
+    # def parse(cls, content: str | int) -> DateTime:
```

Comment:
TODO (I implemented a lot of this in my failed attempt to use the parser to represent a unified unit & date concept)

Reply:
The problem here is that we don't have the original string on the shift node - it can be converted to an integer on parse (e.g. +00010101 becomes 10101, which changes the date definition substantially).

Reply:
Deferring this - as explained in the description.